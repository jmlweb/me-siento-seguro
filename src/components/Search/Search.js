import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import styles from './Search.module.css';
import searchIcon from './search.svg';

class Search extends PureComponent {
  static getDerivedStateFromProps({ initialValue }, { value }) {
    return {
      value: value === undefined ? initialValue : value,
    };
  }

  static defaultProps = {
    initialValue: '',
  };

  static propTypes = {
    title: PropTypes.string.isRequired,
    initialValue: PropTypes.string,
    onSubmit: PropTypes.func.isRequired,
  };

  state = {
    value: undefined,
  };

  onChange = (e) => {
    this.setState({ value: e.target.value });
  };

  handleSubmit = (e) => {
    const { onSubmit } = this.props;
    const { value } = this.state;
    e.preventDefault();
    onSubmit(value);
  };

  render() {
    const { title, initialValue, ...rest } = this.props;
    const { value } = this.state;
    return (
      <div className={styles.wrapper}>
        <form className={styles.form} onSubmit={this.handleSubmit}>
          <p>{title}</p>
          <div className={styles.inputWrapper}>
            <input
              type="text"
              className={styles.input}
              {...rest}
              value={value}
              onChange={this.onChange}
            />
            <button className={styles.button} type="submit">
              <img width="24" height="24" src={searchIcon} alt="search" />
            </button>
          </div>
        </form>
      </div>
    );
  }
}

export default Search;
