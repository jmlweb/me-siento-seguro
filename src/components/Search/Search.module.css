.wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 40px;
}

.form {
  max-width: 600px;
  width: 100%;
  padding: 40px;
  background: rgba(255, 255, 255, 0.5);
  border: 1px solid rgba(0, 0, 0, 0.1);
}

.inputWrapper {
  position: relative;
}

.input {
  display: block;
  width: 100%;
  padding: 15px;
  border-radius: 25px;
  border: 1px solid #bfdbf7;
  outline: 0;
}

.input:focus {
  border: 1px solid #1f7a8c;
}

.button {
  position: absolute;
  right: 48px;
  top: 50%;
  margin-top: -14px;
  appearance: none;
  border: 0;
  background: transparent;
  opacity: 0.5;
}
