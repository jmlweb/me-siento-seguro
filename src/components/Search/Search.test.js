import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Search from './Search';

describe('Search', () => {
  const onSubmit = jest.fn();
  const element = <Search initialValue="" title="Foo" onSubmit={onSubmit} />;
  test('it works', () => {
    const wrapper = shallow(element);
    // component should render
    expect(wrapper).toBeTruthy();
    // value should be initialValue
    expect(wrapper.state('value')).toBe('');
    // value should change for input
    wrapper.find('input').simulate('change', {
      target: {
        value: 'foo',
      },
    });
    expect(wrapper.state('value')).toBe('foo');
    // form should call onSubmit
    wrapper.find('form').simulate('submit', {
      preventDefault: jest.fn(),
    });
    expect(onSubmit).toHaveBeenCalledWith('foo');
  });
  test('snapshot', () => {
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
