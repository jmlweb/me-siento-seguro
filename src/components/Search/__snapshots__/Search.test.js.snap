// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot 1`] = `
<div
  className="wrapper"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <p>
      Foo
    </p>
    <div
      className="inputWrapper"
    >
      <input
        className="input"
        onChange={[Function]}
        onSubmit={
          [MockFunction] {
            "calls": Array [
              Array [
                "foo",
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        type="text"
        value=""
      />
      <button
        className="button"
        type="submit"
      >
        <img
          alt="search"
          height="24"
          src="search.svg"
          width="24"
        />
      </button>
    </div>
  </form>
</div>
`;
